SimpleRefactoring/02_ExtractClassFromParameter.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace CleanCode\SimpleRefactoring;
+
+class ExtractClassFromParameter
+{
+    /**
+     * @param $street
+     * @param $zip
+     * @param $town
+     * @param $region
+     * @param $country
+     * @return string
+     */
+    public function getFullAddress($street, $zip, $town, $region, $country)
+    {
+        return sprintf('%s, %s %s, %s in %s', $street, $zip, $town, $region, $country);
+    }
+
+
+
+
+    /** Refactoring */
+
+    /**
+     * Here can be improved by an Extract Class refactoring from parameter readability:
+     * @param Address $address
+     * @return string
+     */
+    public function getFullAddress2(Address $address)
+    {
+        return sprintf(
+            '%s, %s %s, %s',
+            $address->getStreet(),
+            $address->getZip(),
+            $address->getTown(),
+            $address->getRegion()
+        );
+    }
+
+    /**
+     * Benefits of Using Extract Class from parameter
+     * - more readable to make.
+     * - results in a higher level of abstraction .
+     */
+}
+
+
+
+class Address
+{
+    private $street;
+    private $zip;
+    private $town;
+    private $region;
+    private $country;
+
+    public function __construct($street, $zip, $town, $region, $country)
+    {
+        $this->street = $street;
+        $this->zip = $zip;
+        $this->town = $town;
+        $this->region = $region;
+        $this->country = $country;
+    }
+
+    /**
+     * @return mixed
```

Comment:
all the type hints are incorrect. Do you want to keep them?
I mean they are not the point of this CCD, so it might be ok.